public/app/session-persist.js

```
@@ -1,5 +1,69 @@
+      const raw = sessionStorage.getItem(SESSION_KEY);
+      return raw ? JSON.parse(raw) : {};
+    } catch {
+      return {};
+    }
+  }
+};
+
+// Initialize window.VIP_PARAMS as the loaded store.
+// Existing code that reads window.VIP_PARAMS[key] keeps working.
+if (typeof window !== 'undefined') {
+  if (!window.VIP_PARAMS) window.VIP_PARAMS = SessionPersist.loadAll();
```

ESLint

[error] 'window' is not defined.

(no-undef)